app/src/main/java/dev/baseio/discordjetpackcompose/ui/routes/onboarding/commonui/DiscordDialog.kt

```diff
@@ -0,0 +1,146 @@
+package dev.baseio.discordjetpackcompose.ui.routes.onboarding.commonui
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Divider
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+import dev.baseio.discordjetpackcompose.R
+import dev.baseio.discordjetpackcompose.ui.theme.DiscordDialogTypography
+import dev.baseio.discordjetpackcompose.ui.theme.DiscordJetpackComposeTheme
+import dev.baseio.discordjetpackcompose.ui.theme.discord_dialog_bg
+import dev.baseio.discordjetpackcompose.ui.theme.discord_dialog_button_bg
+import dev.baseio.discordjetpackcompose.ui.theme.discord_dialog_button_row_bg
+import dev.baseio.discordjetpackcompose.ui.theme.discord_dialog_cancel_button_text
+import dev.baseio.discordjetpackcompose.ui.theme.white
+
+@Composable
+fun DiscordDialog(
+    modifier: Modifier = Modifier,
+    show: Boolean,
+    shape: Shape = RectangleShape,
+    backgroundColor: Color = discord_dialog_bg,
+    buttonRowBackgroundColor: Color = discord_dialog_button_row_bg,
+    titleTextProvider: () -> Int,
+    subTitleTextProvider: () -> Int,
+    confirmActionButtonTextProvider: () -> Int,
+    properties: DialogProperties = DialogProperties(),
+    onDismissRequest: () -> Unit,
+    onClickCancelButton: () -> Unit,
+    onClickConfirmActionButton: () -> Unit
+) {
+    if (show) {
+        Dialog(
+            onDismissRequest = { onDismissRequest() },
+            properties = properties
+        ) {
+            Surface(
+                modifier = modifier,
+                shape = shape,
+                color = backgroundColor
+            ) {
+                Column(
```

Comment:
@yugeshjain This UI Tree can be extracted as Composable methods like Title() Subtitle() DialogOptions() DialogPositive() DialogNegative()

Reply:
Done. Resolved